docs/general/run-a-node/maintenance-guides/handling-network-upgrades.md

```diff
@@ -1,81 +1,173 @@
 # Handling Network Upgrades
 
+Changes between the major consensus network versions are backward and forward
+breaking. This means that you need to run a specific version of Oasis Core to
+fetch and validate the blocks matching the specific consensus network version.
+Each network upgrade will have detailed instructions posted in the
+corresponding [Upgrade log]. This guide describes general principles for
+network upgrades and should serve only as a supplementary.
+
 :::caution
 
-Following this guide when there is no network upgrade will result in you losing your place on the current network.
+Stopping and upgrading your node when there is no network upgrade will result
+in losing your place on the current network.
 
 :::
 
-The following guide should be used when the network has agreed to do a software upgrade.
+There are two kinds of consensus network upgrades currently in use:
 
-## Stop the node at specific epoch
+- *Seamless upgrade* requires that you update your Oasis Core without wiping the
+  consensus state or changing the genesis document.
```

Comment:
I think it would make sense to link here to an example upgrade log of a seamless upgrade, e.g.: ```[Testnet upgrade 2022-04-04]: ../../foundation/testnet/upgrade-log#2022-04-04-upgrade)```